src/charm.py

```diff
@@ -345,5 +418,30 @@ def format_labels(label_dict: Dict[str, str]) -> str:
         return ",".join(f"{key}={value}" for key, value in label_dict.items())
 
 
+def generate_provider_name(
+    ingress_app_name: str, ext_authz_info: ProviderIngressConfigData
+) -> str:
+    """Generate a unique provider name from the external authorizer configuration."""
+    data = f"{ext_authz_info.ext_authz_service_name}:{ext_authz_info.ext_authz_port}"
+    stable_hash = hashlib.sha256(data.encode("utf-8")).hexdigest()
+    return f"ext_authz-{ingress_app_name}-{stable_hash}"
```

Comment:
I am not 100% sure but I think `"ext_authz-{ingress_app_name}"` should be guaranteed to be unique. Perhaps we do not need the hash.

Reply:
we discussed this offline and @dstathis clarified that this will be unique because either:
* the relation is in model, thus the app name is unique
* the relation is cross model, thus this is the sas name not the related app's name, and the sas name is always unique

So +1 to dylan's request!